docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +259,152 @@ it may still remain unpaid if the pending amounts fail to process.
+- `canceledAmount` - sum of `CANCEL_SUCCESS` event amounts
+- `cancelPendingAmount` - sum of `CANCEL_REQUEST` event amounts
+
+
+:::note
+`AUTHORIZATION_ACTION_REQUIRED`, `CHARGE_ACTION_REQUIRED`, and `INFO` event types are not included in amount calculations.
+:::
+
+To learn more about each event type, see [Transaction event types](#transaction-event-types) section.
+
+#### Events deduplication
+
+Each event in `TransactionItem` is identified by `pspReference`. This field is used to de-duplicate events in case there were reported multiple times by the payment app. Deduplication is done within single `TransactionItem`.
+
+For example if Saleor receives two `CHARGE_SUCCESS` events for a single `TransactionItem` with the same `pspReference` and `amount` there will be only one event stored on the Transaction. Second event will be treated as "already reported" and it won't be stored by Saleor.
+
+:::caution
+If the `CHARGE_SUCCESS` event has different `amount` Saleor **will throw an error**.
+:::
+
+When reporting such events in `transactionEventReport` mutation it returns event that already exists and a flag `alreadyReported: true`.
+
+You can learn more about this mutation and event deduplication in [Reporting actions for transactions](/developer/payments/transactions.mdx#reporting-actions-for-transactions) docs.
+
+### State machine
+
+Saleor Transaction model **is not a state machine**. This allows great flexibility when integrating with many payment providers, even those that might implement some unconventional flows.
+However this also means that we do not forbid changes on Transactions that might be seen as invalid by your business use-case, such validation is on the Payment App side.
+
+For example:
+- Available `actions` on `TransactionItem` (list of `REFUND`, `CHARGE`, `CANCEL`) are honored by Saleor Dashboard, but are not rejected once `transactionRequestAction` is requested via API - such request is sent to payment app
+- Transaction might receive `CANCEL_SUCCESS` event even though it had `CHARGE_SUCCESS` reported (in such case to lower `chargedAmount` app would need to report `REFUND_SUCCESS` event).
+
+#### Existing validation
+
+With that in mind, Saleor prohibits following changes, that would otherwise cause incorrect internal calculations:
+- Multiple `AUTHORIZATION_SUCCESS` events with different either `pspReference` or `amount`
+  - Saleor will throw an exception: "Event with `AUTHORIZATION_SUCCESS` already reported for the transaction. Use `AUTHORIZATION_ADJUSTMENT` to change the authorization amount."
+
+:::note
+**Example**: `AUTHORIZATION_SUCCESS` event with `pspReference: psp1` and `amount: 10 USD`, then report `AUTHORIZATION_SUCCESS` event with `pspReference: psp2` and `amount: 10 USD`
+:::
+
+:::tip
+**Fix**: In order to change `authorizedAmount` value, payment app should send `AUTHORIZATION_ADJUSTMENT` event. If there are in fact two valid `AUTHORIZATION_SUCCESS` events, create new `TransactionItem`
```

Comment:
should app create new `TransactionItem` or Saleor will do it automatically?

Reply:
Nope, this has to be done by app - Saleor will reject `transactionEventReport` mutation if app would report 2 `AUTHORIZATION_SUCCESS` events